src/main/java/frc/robot/subsystems/SwerveDrive.java

```diff
@@ -385,17 +396,23 @@ private double modifyAxis(double value, boolean squared) {
                 return value;
         }
 
+        public RunCommand xModeCommandFactory() {
+                return new RunCommand(
+                       () -> {
+                        SwerveModuleState[] states = {
+                                new SwerveModuleState(0, Rotation2d.fromDegrees(135)), 
+                                new SwerveModuleState(0, Rotation2d.fromDegrees(45)), 
+                                new SwerveModuleState(0, Rotation2d.fromDegrees(45)), 
+                                new SwerveModuleState(0, Rotation2d.fromDegrees(135))};
+                                setModuleStates(states);
+                       } 
+                );
+                
+        }
+
         @Override
         public void periodic() {
-                for (SwerveModule module : m_modules) {
-                        module.zeroModule();
-                }
```

Comment:
These lines being deleted is probably why the robot wasn't driving correctly. This is my fix to the cancoder initialization bug.

Reply:
fixed by only calling the zero when disabled